File: src/uv_upx/services/updater/update_lock_file.py
from typing import TYPE_CHECKING

from safe_result import safe_with

from uv_upx.services.local_segments.exceptions import NonEmptyLocalSegmentsError
from uv_upx.services.run_uv_related import UnresolvedDependencyError, run_uv_lock

if TYPE_CHECKING:
    import pathlib


@safe_with(UnresolvedDependencyError, NonEmptyLocalSegmentsError)
def update_lock_file(
    project_root_path: pathlib.Path,
) -> None:
    # Because we want a fast update. Without triggering build for now.
    run_uv_lock(
        workdir=project_root_path,
        upgrade=True,
    ).unwrap()

File: rust/uv_upx_r/src/services/updater/update_lock_file.rs
use crate::services::run_uv_related::run_uv_lock::run_uv_lock;

pub async fn update_lock_file(
    project_root_path: std::path::PathBuf,
) -> Result<(), Box<dyn std::error::Error>> {
    // Because we want a fast update. Without triggering build for now.
    run_uv_lock(
        &project_root_path,
        true,
    )
    .await?;

    Ok(())
}